### PR TITLE
apps: add basic components for the new list of cluster apps

### PR DESCRIPTION
**Pull request by axbarsan**
Towards https://github.com/giantswarm/giantswarm/issues/18522

This PR includes:
- feature flag for enabling the new UI (disabled by default)
- basic components for the list of apps (only include the name and version when collapsed)

**Note: This doesn't affect the existing UI in any way. Only GS staff can enable the `Use NextGen cluster apps UI` experiment and see the new UI.**

## Preview 

<details>
<summary>Just chillin</summary>

image

</details>

<details>
<summary>App open</summary>

image

</details>

<details>
<summary>No apps</summary>

image

</details>

<details>
<summary>Error loading apps</summary>

image

</details>

<details>
<summary>Loading apps</summary>

image

</details>

### CODE REVIEW

File: src/components/MAPI/apps/ClusterDetailAppList.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+const LOADING_COMPONENTS = new Array(3).fill(0);
+
+interface IClusterDetailAppListProps
+  extends React.ComponentPropsWithoutRef<typeof Box> {
+  apps: applicationv1alpha1.IApp[];
+  isLoading?: boolean;
+  errorMessage?: string;
+}
+
+const ClusterDetailAppList: React.FC<IClusterDetailAppListProps> = ({
```
Comment (axbarsan) [inline src/components/MAPI/apps/ClusterDetailAppList.tsx:16]: The juicy bit #1

File: src/components/MAPI/apps/ClusterDetailAppListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  transition: 0.15s ease-out;
+`;
+
+interface IClusterDetailAppListItemProps
+  extends React.ComponentPropsWithoutRef<typeof Box> {
+  app?: applicationv1alpha1.IApp;
+  isActive?: boolean;
+}
+
+const ClusterDetailAppListItem: React.FC<IClusterDetailAppListItemProps> = ({
```
Comment (axbarsan) [inline src/components/MAPI/apps/ClusterDetailAppListItem.tsx:19]: The juicy bit #2

File: src/components/MAPI/apps/ClusterDetailApps.tsx [hunk 3]
```diff
@@ -212,29 +212,53 @@ const ClusterDetailApps: React.FC<IClusterDetailApps> = ({
...
-            }))}
-            error={extractErrorMessage(appListError) ?? null}
-            onShowDetail={showAppDetail}
-          >
-            <BrowseButtonContainer>
-              <BrowseButton
-                onClick={openAppCatalog}
-                disabled={appListIsLoading}
-                icon={<i className='fa fa-add-circle' />}
+          {featureFlags.flags.NextGenClusterApps.enabled ? (
```
Comment (axbarsan) [inline src/components/MAPI/apps/ClusterDetailApps.tsx:215]: If the feature flag is enabled, we display the new UI

**Review by kuosandys — APPROVED** [review]
Looks great!

**Pull request merged**